src/mrpro/algorithms/optimizers/pgd.py

```diff
@@ -0,0 +1,158 @@
+
+    For fixed stepsize t, pgd converges globally when :math:`t \in (0, 1/L)`,
+    where L is the Lipschitz constant of the gradient of f.
+    In applications, f is usually of the form :math:`f(x) = 1/2 ||Ax - target||^2`,
+    where :math:`A` is a linear operator.
+    In this case, :math:`t \in (0, 1/||A^T A||)` for convergence.
+    If no backtracking is used, the fixed stepsize should be given accordingly to the convergence condition.
+
+    Example:
+        L1 regularized image reconstruction. Problem formulation: :math:`min_x 1/2 ||Fx - target||^2 + ||x||_1`,
+        with :math:`F` being the Fourier Transform, target the acquired data \in k-space and x \in image space,
+        :math:`f(x) = 1/2 ||Fx - target||^2, g(x) = ||x||_1`.
+        In this case, :math:`||F^T F|| = 1`. ::
+                fft = FastFourierOp()
+                l2 = L2NormSquared(target=kspace_data)
+                f = l2 @ fft
+                g = L1Norm()
+                fft_norm = 1.
+                stepsize = 0.85 * 1 / fft_norm
+                initial_value = torch.ones(image_space_shape)
+                pgd_image_solution = pgd(
+                    f=f,
+                    g=g,
+                    initial_value=initial_value,
+                    stepsize=stepsize,
+                    reg_parameter=0.01,
+                    max_iterations=200,
+                    backtrack_factor=1.0,
+                )
+
+
+    Parameters
+    ----------
+    f
+        convex, differentiable functional
+    g
+        convex, non-smooth functional with computable proximal map
+    initial_value
+        initial value for the solution of the algorithm
+    stepsize
+        stepsize needed in the gradient step, constant if no backtracking is used, otherwise it is the initial stepsize
+    reg_parameter
+        regularization parameter that multiplies g
+    max_iterations
+        number of iterations
+    backtrack_factor
+        must be :math:`<=1`. if :math:`<1.`, backtracking rule for stepsize following https://www.ceremade.dauphine.fr/~carlier/FISTA
+        is used
+    callback
+        function to be called at each iteration
+
+    Returns
+    -------
+        an approximate solution of the minimization problem
+    """
+    # check that backtracking factor is in the correct range
+    if not 0.0 <= backtrack_factor <= 1.0:
+        raise ValueError('Backtracking factor must be in the range [0, 1].')
+
+    backtracking = not math.isclose(backtrack_factor, 1)
+
+    if isinstance(initial_value, torch.Tensor):
+        initial_values: tuple[torch.Tensor, ...] = (initial_value,)
+    else:
+        initial_values = initial_value
+
+    if isinstance(g, ProximableFunctional):
+        g_sum = ProximableFunctionalSeparableSum(g)
+    else:
+        g_sum = g
+
+    x_old = initial_values
+    y: tuple[torch.Tensor, ...] = initial_values
+    gradient: tuple[torch.Tensor, ...]
+    f_y: torch.Tensor
+    t_old = 1.0
```

Comment:
of we set this to 0, we get ISTA?

Reply:
you mean if `t_old=0`? If so I wouldn't say so:
if we start with t_1 = 0, y_1=x_0

then at iteration=1:
x_1 = prox(y_1)
t_2 = 1
y_2 = x_1 - (x_1 - x_0) = x_0

and so at iteration=2: 
x_2 = prox(x_0) ?? (would expect x_1 if it was ista)
and t_3 != 1 

or not? maybe it's wrong but I would say it needs to be coded as separate cases if we want both ISTA and FISTA?

Reply:
or did you mean setting something else to 0?

Reply:
final dicision: only use FISTA